### PR TITLE
ECP: Add Unit Tests for secp256k1

### LINKED ISSUE #7258: Add unit tests for ecp_mod_p256k1()
The task is to add unit tests for ecp_mod_p256k1(). This function reduces the input modulo secp256k1_p. The result of ecp_mod_p256k1() is not always canonical, use mpi_mod_mpi() to get a canonical representative for easier testing.

Only cases where the input is non-negative and at most twice as long as the modulus. (This is because soon these will be replaced by their fixed width _raw counterparts.)

### PR DESCRIPTION
**Pull request by minosgalanakis**
## Description
Resolves #7258

## Gatekeeper checklist

- [x] **changelog** not required. Extending testing coverage, does not affect library's behaviour
- [x] **backport** not required. This is worked required by a newly developped interface
- [x] **tests** provided,.

### CODE REVIEW

**Comment by yanesca** [comment]
@minosgalanakis this is a deliverable for this quarter, it should have a `priority-high` label, could you please add it?

**Comment by minosgalanakis** [comment]
As before it is based on PR #7425 . Please review only [last 3 commits](https://github.com/Mbed-TLS/mbedtls/pull/7427/files/357b9e1342f32f507d82c097beeeb656a117c23a..f2568cad2347d530eb8180def0083f337d131da0) if need to. 

It will rebased after #7425 has been merged.

**Comment by minosgalanakis** [comment]
PR Rebased on latest development. Ready to review

File: library/ecp_curves.c [hunk 1]
```diff
@@ -4618,6 +4618,8 @@ int mbedtls_ecp_mod_p224k1(mbedtls_mpi *);
 #endif
 #if defined(MBEDTLS_ECP_DP_SECP256K1_ENABLED)
 static int ecp_mod_p256k1(mbedtls_mpi *);
+MBEDTLS_STATIC_TESTABLE
+int ecp_mod_p256k1(mbedtls_mpi *);
```
Comment (paul-elliott-arm) [inline library/ecp_curves.c:4622]: Shouldn't this be:

```suggestion
int mbedtls_ecp_mod_p256k1(mbedtls_mpi *);
```

**Review by paul-elliott-arm — CHANGES_REQUESTED** [review]
1 thing jumped out. Everything else looks ok, but need to do a slightly deeper pass.

File: scripts/mbedtls_dev/ecp.py [hunk 2]
```diff
@@ -617,3 +617,67 @@ def result(self) -> List[str]:
     @property
     def is_valid(self) -> bool:
         return True
+
```
Comment (gabor-mezei-arm) [inline scripts/mbedtls_dev/ecp.py:620]: Before class definition there should be two empty line.

**Review by gabor-mezei-arm — APPROVED** [review]
Only a minor non-blocking issue found otherwise looks good to me.

**Pull request merged**